### PR TITLE
Create new setting window

### PR DESCRIPTION
**Pull request by ssmr56**

### CODE REVIEW

**Comment by ssmr56** [comment]
@0leksandrBondar pls, provide review

**Comment by 0leksandrBondar** [comment]
CR

File: source/CMakeLists.txt [hunk 1]
```diff
@@ -35,9 +36,11 @@ set(HEADERS
     model/cardmodel.h
     model/cardsuit.h
     model/playermodel.h
+    dialogWindows/settingwindow.h
 )
 
-add_executable(${PROJECT_NAME} ${SOURCES} ${HEADERS})
+add_executable(${PROJECT_NAME} ${SOURCES} ${HEADERS}
+    dialogWindows/settingwindow.h dialogWindows/settingwindow.cpp)
```
Comment (0leksandrBondar) [inline source/CMakeLists.txt]: remove this

File: source/graphicsItems/gamewidget.cpp [hunk 2]
```diff
@@ -33,7 +34,7 @@ GameWidget::GameWidget(QWidget *parent)
       _soundButton{ new QPushButton(QStringLiteral("sound"), this) },
       _labelsLayout{ new QHBoxLayout(this) },
       _settingsButton{ new QPushButton(QStringLiteral("setting"), this) },
-      _settingDialog{ new QDialog() }
+      _settingWindow{ new SettingWindow() }
```
Comment (0leksandrBondar) [inline source/graphicsItems/gamewidget.cpp]: add "this" as parameter

File: source/dialogWindows/settingwindow.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <QDialog>
+
+class SettingWindow : public QDialog
+{
+    Q_OBJECT
+public:
+    SettingWindow();
```
Comment (0leksandrBondar) [inline source/dialogWindows/settingwindow.h]: add "QWidget *parent = nullptr" as parametr in constructor

File: source/dialogWindows/settingwindow.cpp [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#include "settingwindow.h"
+
+SettingWindow::SettingWindow() {}
```
Comment (0leksandrBondar) [inline source/dialogWindows/settingwindow.cpp]: SettingWindow::SettingWindow("QWidget *parent) : QDialog(parent){}

File: source/graphicsItems/gamewidget.cpp [hunk 5]
```diff
@@ -42,12 +43,10 @@ GameWidget::GameWidget(QWidget *parent)
     setPalette(pal);
     setGridUi();
 
-    connect(_settingsButton.get(), &QPushButton::clicked, this,
-            &GameWidget::makeSettingDialogWindow);
+    connect(_settingsButton.get(), &QPushButton::clicked, _settingWindow.get(),
```
Comment (0leksandrBondar) [inline source/graphicsItems/gamewidget.cpp]: remove .get() for _settingWindow

File: source/graphicsItems/gamewidget.h [hunk 6]
```diff
@@ -37,5 +37,5 @@ class GameWidget : public QWidget
     std::shared_ptr<QPushButton> _soundButton;
     std::shared_ptr<QHBoxLayout> _labelsLayout;
     std::shared_ptr<QPushButton> _settingsButton;
-    std::unique_ptr<QDialog> _settingDialog;
+    std::unique_ptr<SettingWindow> _settingWindow;
```
Comment (0leksandrBondar) [inline source/graphicsItems/gamewidget.h]: remove std::unique_ptr

**Comment by 0leksandrBondar** [comment]
inital CR done

**Comment by ssmr56** [comment]
Provide review again

File: source/dialogWindows/settingwindow.cpp [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#include "settingwindow.h"
+
+SettingWindow::SettingWindow(QWidget *parent) : QDialog(parent) {}
+
+void SettingWindow::makeSettingDialogWindow() { exec(); }
```
Comment (0leksandrBondar) [inline source/dialogWindows/settingwindow.cpp]: please rename this method "makeSettingDialogWindow" -> "showSettingsWindow"

File: source/graphicsItems/gamewidget.cpp [hunk 8]
```diff
@@ -1,8 +1,9 @@
 #include "gamewidget.h"
 
+#include "dialogWindows/settingwindow.h"
 #include "gameController/gameController.h"
 
-#include <QDialog>
+// #include <QDialog>
```
Comment (0leksandrBondar) [inline source/graphicsItems/gamewidget.cpp]: remove

**Comment by ssmr56** [comment]
I finished with it, check again

File: source/graphicsItems/gamewidget.cpp [hunk 9]
```diff
@@ -42,12 +42,10 @@ GameWidget::GameWidget(QWidget *parent)
...
     setGridUi();
 
-    connect(_settingsButton.get(), &QPushButton::clicked, this,
-            &GameWidget::makeSettingDialogWindow);
+    connect(_settingsButton.get(), &QPushButton::clicked, _settingWindow,
+            &SettingWindow::showSettingsWindow);
 }
 
-void GameWidget::makeSettingDialogWindow() { _settingDialog->exec(); }
-
```
Comment (0leksandrBondar) [inline source/graphicsItems/gamewidget.cpp:50]: why do you deleted this space?)

**Comment by 0leksandrBondar** [comment]
CR Done

**Pull request merged**